commands/utils/TerminalProvider.ts

```diff
@@ -11,15 +11,32 @@ export interface ITerminalProvider {
 }
 
 export class DefaultTerminalProvider {
+  private _cwd: vscode.Uri | undefined;
+
+  constructor(extensionExecutionContext: vscode.ExtensionExecutionContext) {
+    // When the extension is executing locally but there is a remote present,
```

Comment:
Is there any documentation for this behavior? I want to make sure this is actually an intended design choice for how terminals work in VS Code - rather than a random "bug" (aka "feature") that we should be relying on